File: cpp/easy/0118_pascals_triangle.cpp
// https://leetcode.com/problems/pascals-triangle/description/
class Solution {
public:
    vector<int> getNextRow(vector<int> prev){
        vector<int> row;
        row.push_back(1);
        for(int i = 0; i < prev.size() - 1; i++){
            row.push_back(prev[i] + prev[i + 1]);
        }
        row.push_back(1);

        return row;
    }

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> tri;
        tri.push_back({1});
        for(int i = 0; i < numRows - 1; i++){
            tri.push_back(getNextRow(tri[i]));
        }
        return tri;
    }
};

File: python/easy/0118_pascals_triangle.py
# https://leetcode.com/problems/pascals-triangle/description/?envType=daily-question&envId=2025-08-01
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        def getNextRow(prev):
            row = [1]
            for i in range(len(prev) - 1):
                row.append(prev[i] + prev[i + 1])
            row.append(1)
            return row

        triangle = [[1]]
        for i in range(numRows - 1):
            triangle.append(getNextRow(triangle[i]))
        return triangle
